### DISCUSSION TITLE
limit voucher used to once per user

### DISCUSSION THREAD

**Discussion by DigitalTrading** [openstore-ecommerce/OpenStore#262] [General]
Hi,
I have created a voucher code which I want to have unlimited usage but just once per user. All purchases require the user to be logged in so it should be possible.
Thanks

**Comment by DNNMonster** [comment]
I do not believe this is possible from a default install.   I think a plugin can solve the problem but if there is agreement that it should be a part of the default install then it can be addressed.  It's up to @leedavi

**Comment by leedavi** [comment]
The promotions are a core plugin.  Our current policy of support is no feature change in the core system and plugins.

If a change was to be made on this core pluigin, backward compatibility testing would be needed.   Which would mean it'll be quicker copying this plugin, renaming and adding the new functionality.

At the moment, If you want one of the team to do this work, it would need to be sponsored.

**Comment by DigitalTrading** [comment]
Thanks for the feedback, I may try and develop a plugin to support this. I will let you know how I get on.

**Comment by leedavi** [comment]
OK Great. Send me a email if you need some advice.

Here are the documentation files, there a little old now, so don't read them as a bible: 
https://doc.openstore-ecommerce.com/Developer-Guide

I think your best option would be to take a copy of the "Promotions" plugin from the core and alter it to add the extra functionality.

https://github.com/openstore-ecommerce/OpenStore/tree/master/Providers/PromoProvider

**Comment by leedavi** [comment]
https://github.com/openstore-ecommerce/Developer-Documents/blob/master/CreatePluginSingleInterface.pdf

https://github.com/openstore-ecommerce/OS_PluginTemplateSI
